scheduler/test/cook/test/mesos/scheduler.clj

```diff
@@ -836,6 +836,28 @@
                      (make-dummy-status-update task-id-a :mesos-slave-restarted :task-running)))
         (is (true? (contains? @tasks-killed task-id-a)))
         ))
+
+    (testing "instance persists mesos-start-time when task is first known to be starting or running"
+      (let [job-id (create-dummy-job conn
+                                     :user "mforsyth"
+                                     :job-state :job.state/running
+                                     :retry-count 3)
+            task-id "task-mesos-start-time"
+            instance-id-a (create-dummy-instance conn job-id
+                                                 :instance-status :instance.status/unknown
+                                                 :task-id task-id
+                                                 :reason :unknown)
+            mesos-start-time (fn [] (-> conn
+                                        d/db
+                                        (d/entity [:instance/task-id task-id])
+                                        :instance/mesos-start-time))]
+        (is (nil? (mesos-start-time)))
+        (async/<!! (sched/handle-status-update conn driver fenzo
+                                               (make-dummy-status-update task-id :unknown :task-staging)))
+        (is (nil? (mesos-start-time)))
+        (async/<!! (sched/handle-status-update conn driver fenzo
+                                               (make-dummy-status-update task-id :unknown :task-running)))
+        (is (not (nil? (.getTime (mesos-start-time)))))))
```

Comment:
Maybe send another status update after this and assert that the persisted time doesn't change?